goinstant/docker_test.go

```diff
@@ -0,0 +1,948 @@
+				return 0, nil
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			OsCreate = tt.osCreate
+			IoCopy = tt.ioCopy
+
+			if err := createZipFile(tt.args.file, tt.args.content); (err != nil) != tt.wantErr {
+				t.Errorf("createZipFile() error = %v, wantErr %v", err, tt.wantErr)
+				log.Println(tt.name, "failed!")
+			} else {
+				log.Println(tt.name, "passed!")
+			}
+		})
+	}
+}
+
+func Test_runCommand(t *testing.T) {
+	testCases := []struct {
+		commandName     string
+		suppressErrors  []string
+		commandSlice    []string
+		pathToPackage   string
+		errorString     error
+		testInfo        string
+		execCommandMock func(commandName string, commandSlice ...string) *exec.Cmd
+	}{
+		{
+			commandName:     "docker",
+			suppressErrors:  nil,
+			commandSlice:    []string{"ps"},
+			pathToPackage:   "",
+			errorString:     nil,
+			testInfo:        "runCommand - run basic docker ps test",
+			execCommandMock: exec.Command,
+		},
+		{
+			commandName:     "docker",
+			suppressErrors:  nil,
+			commandSlice:    []string{"volume", "rm", "test-volume"},
+			pathToPackage:   "",
+			errorString:     fmt.Errorf("Error waiting for Cmd. Error: No such volume: test-volume\n: exit status 1"),
+			testInfo:        "runCommand - removing nonexistant volume should return error",
+			execCommandMock: exec.Command,
+		},
+		{
+			commandName:     "docker",
+			suppressErrors:  []string{"Error: No such volume: test-volume"},
+			commandSlice:    []string{"volume", "rm", "test-volume"},
+			pathToPackage:   "",
+			errorString:     nil,
+			testInfo:        "runCommand - error thrown should be suppressed",
+			execCommandMock: exec.Command,
+		},
+		{
+			commandName:    "git",
+			suppressErrors: nil,
+			commandSlice:   []string{"clone", "git@github.com:testhie/test.git"},
+			pathToPackage:  "test",
+			errorString:    nil,
+			testInfo:       "runCommand - clone a custom package and return its location",
+			execCommandMock: func(commandName string, commandSlice ...string) *exec.Cmd {
+				cmd := exec.Command("pwd")
+				return cmd
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.testInfo, func(t *testing.T) {
+			execCommand = tt.execCommandMock
+			pathToPackage, err := runCommand(tt.commandName, tt.suppressErrors, tt.commandSlice...)
+			if !assert.Equal(t, pathToPackage, tt.pathToPackage) {
+				t.Fatal("RunCommand failed - path to package returned is incorrect " + pathToPackage)
+			}
+			if err != nil && tt.errorString != nil && !assert.Equal(t, err.Error(), tt.errorString.Error()) {
+				t.Fatal("RunCommand failed - error returned incorrect")
+			}
+
+			if (err != nil && tt.errorString == nil) || (err == nil && tt.errorString != nil) {
+				log.Fatal("RunCommand failed - error returned incorrect")
+			}
+
+			t.Log(tt.testInfo + " passed!")
+		})
+	}
+}
+
+func Test_mountPackage(t *testing.T) {
+	defer gock.Off()
+	defer resetMountPackageMocks(runCommand, unzipPackage, untarPackage)
```

Comment:
Not quite sure what you're trying to do here?

If it's resetting the variables for use outside of testing, shouldn't we be doing that everywhere? <-- this obviously depends on the plan for running these tests.

Reply:
The other tests for functions that also use the functionality that is mocked were failing. The mocked functions have to be reset